pandas/core/arrays/base.py

```diff
@@ -1167,6 +1167,10 @@ class ExtensionOpsMixin:
        with NumPy arrays.
     """
 
+    @classmethod
+    def _create_arithmetic_method(cls, op):
```

Comment:
I recall that replacing
```
cls.__add__ = cls._create_arithmetic_method(operator.add)
```
etc. with
```
setattr(cls, "__add__",  cls._create_arithmetic_method(operator.add))
```

solves mypy issues, although I'm not getting failures if I remove this classmethod added here, so not sure what the mypy error is here.

which way do you prefer?

Reply:
i prefer the aesthetics of the `cls.__add__ =` variant, but not a strong enough preference to make a fuss over

Reply:
sure